source/extensions/filters/http/jwt_authn/filter_factory.cc

```diff
@@ -45,9 +45,22 @@ FilterFactory::createFilterFactoryFromProtoTyped(const JwtAuthentication& proto_
                                                  Server::Configuration::FactoryContext& context) {
   validateJwtConfig(proto_config);
   auto filter_config = std::make_shared<FilterConfig>(proto_config, prefix, context);
-  return [filter_config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+
+  auto factory = [filter_config](const std::vector<std::string>& audiences,
+                                 const absl::optional<std::string>& issuer,
+                                 std::vector<JwtLocationConstPtr>&& tokens) -> AuthenticatorPtr {
+    return Authenticator::create(audiences, issuer, std::move(tokens), filter_config);
+  };
+
+  std::vector<MatcherConstSharedPtr> rule_matchers;
```

Comment:
This rule_matchers can be part of FilterConfig.

Reply:
I need to create authenticators in verifiers, and authenticator depends on filter config to function. If I move matchers to FilterConfig. Wouldn't this create a circular dependency? I not sure how to get around that.

Reply:
Authenticator only need JwksCache, not the whole FilterConfig

Reply:
FilterConfig also owns cluster manager.

Reply:
We only need one object to hold all global data.  FilterConfig is that object.  We should not have multiple global objects scatter around.